# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/actors/BrowserTabChild.jsm
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc browser/actors/BrowserTabChild.jsm
--- a/browser/actors/BrowserTabChild.jsm	Wed Apr 28 11:09:38 2021 +0000
+++ b/browser/actors/BrowserTabChild.jsm	Wed Apr 28 11:26:49 2021 +0000
@@ -85,6 +85,10 @@
         } catch (e) {}
 
         let reloadFlags = message.data.flags;
+        if (message.data.handlingUserInput) {
+          reloadFlags |= Ci.nsIWebNavigation.LOAD_FLAGS_USER_ACTIVATION;
+        }
+
         try {
           E10SUtils.wrapHandlingUserInput(
             this.document.defaultView,